src/main/java/de/qabel/desktop/storage/command/CreateFolderChange.java

```diff
@@ -0,0 +1,29 @@
+package de.qabel.desktop.storage.command;
+
+import de.qabel.core.crypto.CryptoUtils;
+import de.qabel.desktop.exceptions.QblStorageException;
+import de.qabel.desktop.storage.BoxFolder;
+import de.qabel.desktop.storage.DirectoryMetadata;
+import org.spongycastle.crypto.params.KeyParameter;
+
+public class CreateFolderChange implements DirectoryMetadataChange<ChangeResult<BoxFolder>> {
+	private CryptoUtils cryptoUtils = new CryptoUtils();
+	private String name;
+	private byte[] deviceId;
+	private final KeyParameter secretKey;
+
+	public CreateFolderChange(String name, byte[] deviceId) {
+		this.name = name;
+		this.deviceId = deviceId;
+		secretKey = cryptoUtils.generateSymmetricKey();
```

Comment:
On conflict, the `execute()` method is executed at least twice (once on initial createFolder and the second time when it's reapplied to the reloaded DM).
If we use another key, the new (already uploaded) DM can't be read / decrypted, can it?